### PR TITLE
docker compose added; datasource and dashboard are provisioned

### PR DESCRIPTION
**Pull request by vmichalkevicius-rocket**
- docker compose with prometheus and grafana created
- prometheus datasource provisioned on startup
- sample ODP Connect dashboard provisioned on startup

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+services:
+ prometheus:
+   image: prom/prometheus:v2.53.1
+   container_name: prometheus
+   ports:
+     # port number on the left hand side can be changed to other value if 9090 is beeing in use
```
Comment (ghannington-Rocket) [inline docker-compose.yaml]: Delete "beeing"

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+     retries: 10
+     start_period: 40s
+   restart: unless-stopped
+ grafana:
+   image: grafana/grafana-oss:11.1.3
+   container_name: grafana
+   depends_on:
+     - prometheus
+   ports:
+     # port number on the left hand side can be changed to other value if 3000 is beeing in use
```
Comment (ghannington-Rocket) [inline docker-compose.yaml]: Delete "beeing"

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+     - 3000:3000
+   volumes:
+     - grafana-data:/var/lib/grafana
+     # provisioning configs
+     - ./config/grafana/provisioning:/etc/grafana/provisioning
+     # dashboards to privision
```
Comment (ghannington-Rocket) [inline docker-compose.yaml]: Typo: replace "pr**i**vision" with "pr**o**vision"

File: readme.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+Once all that is done, restart Data Broker for new settings to take effect.
+
+At that point, both Data Connect and Data Broker metrics (`odp_server_*` and `odp_broker_*`) will be available on your Prometheus endpoint.
+
+# Installation & Configuration
+Clone the project repository.
+
+There are two ways of how you can you use provided dashboard:
+- in existing Prometheus/Grafana environment
+- in dedicated docker based environment
```
Comment (ghannington-Rocket) [inline readme.md]: Nit: in prose (as opposed to, for example, commands), when referring to the Docker platform, refer to "Docker" with a capital D, not "docker".

File: readme.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+KAY.CIDB.FWD.ST.SOURCE_STORE=STATS                      
+KAY.CIDB.FWD.ST.SINK_HOST=[DATA_CONNECT_HOST]   
+KAY.CIDB.FWD.ST.SINK_PORT=[DATA_CONNECT_PORT]
+```
+Once all that is done, restart Data Broker for new settings to take effect.
+
+At that point, both Data Connect and Data Broker metrics (`odp_server_*` and `odp_broker_*`) will be available on your Prometheus endpoint.
+
+# Installation & Configuration
+Clone the project repository.
```
Comment (davdai01) [inline readme.md]: This line should probably be moved to the `## Dedicated Docker setup` section, as this is not needed for `## Existing Prometheus/Grafana setup`

File: config/prometheus/prometheus.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+global:
+  scrape_interval: 15s
+
+scrape_configs:
+  - job_name: 'odp-connect-stats'
+    scrape_interval: 5s
```
Comment (davdai01) [inline config/prometheus/prometheus.yaml]: This line is probably not needed. `5s` is unnecessary. The default `15s` should be fine.

**Review by davdai01 — CHANGES_REQUESTED** [review]

**Review by davdai01 — APPROVED** [review]

**Pull request merged**